### PR TITLE
Avoid fail fetch when no apiKey

### PR DESCRIPTION
**Pull request by edellucien**
Mobility-toolbox-js updated for apiKey handling
Prevent to request without apiKey defined
Introduce new param to MapboxStyleLayer, to be bale to switch MapboxLayer style.

# Others

- [ ] It's not a hack or at least an unauthorized hack :).
- [ ] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [ ] The author of the MR has made its own review before assigning the reviewer.
- [ ] IE11 tested.
- [x] The title means something for a human being.
- [x] The title contains [WIP] if it's necessary.
- [x] Labels applied. if it's a release? a hotfix?
- [ ] Tests added.

### CODE REVIEW

File: src/components/TopicLoader/TopicLoader.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const propTypes = {
     push: PropTypes.func,
     replace: PropTypes.func,
   }),
-  apiKey: PropTypes.string.isRequired,
+  apiKey: PropTypes.string,
   apiKeyName: PropTypes.string.isRequired,
```
Comment (oterral) [inline src/components/TopicLoader/TopicLoader.js:26]: apiKeyName is also not required and default value is 'key'

Comment (edellucien) [reply src/components/TopicLoader/TopicLoader.js]: Indeed I adapt it

File: src/layers/TrafimageMapboxLayer/TrafimageMapboxLayer.js [hunk 2]
```diff
@@ -66,15 +67,29 @@ class TrafimageMapboxLayer extends MapboxLayer {
...
     }
-    const { style } = this.options;
-    const newStyleUrl = `${url}/styles/${style}/style.json${
-      key ? `?${apiKeyName || 'key'}=${key}` : ''
-    }`;
 
+    const newStyleUrl =
+      this.styleUrl && this.style
+        ? this.createStyleUrl()
+        : `${this.url}/styles/${this.style}/style.json?`;
```
Comment (oterral) [inline src/layers/TrafimageMapboxLayer/TrafimageMapboxLayer.js:92]: Should  it be 
this.styleUrl = this.createStyleUrl(this.apiKey, this.apiKeyName , this.styleUrl); 
?

Comment (oterral) [inline src/layers/TrafimageMapboxLayer/TrafimageMapboxLayer.js:92]: Should it be
this.styleUrl = this.createStyleUrl();
?

File: src/layers/TrafimageMapboxLayer/TrafimageMapboxLayer.js [hunk 3]
```diff
@@ -66,15 +67,29 @@ class TrafimageMapboxLayer extends MapboxLayer {
...
+    this.style = newStyle || this.options.style;
+    this.setStyleConfig();
+  }
+
   setStyleConfig(url, key, apiKeyName) {
-    if (!url) {
-      return;
+    if (!url && !key && !apiKeyName) {
+      this.styleUrl = `${this.url}/styles/${this.style}/style.json?`;
+    }
```
Comment (oterral) [inline src/layers/TrafimageMapboxLayer/TrafimageMapboxLayer.js:78]: I don't think this if is useful

Comment (edellucien) [reply src/layers/TrafimageMapboxLayer/TrafimageMapboxLayer.js]: Code cleared and updated

**Review by oterral — CHANGES_REQUESTED** [review]
lgtm

**Review by oterral — APPROVED** [review]

**Pull request merged**